portal-impl/src/com/liferay/portal/util/HttpImpl.java

```diff
@@ -1262,22 +1262,18 @@ public String shortenURL(String url, int count) {
 					sb.append(qName);
 					sb.append(StringPool.EQUAL);
 					sb.append(newURL);
-
-					if (i < (params.length - 1)) {
```

Comment:
This does not look right. Since this StringPool.AMPERSAND append is inside a nested if, it does not always run. But outside the for loop, you always cut off the last char. What if the last iteration was inside the outer if, but not inside the inner if. I don't think you want to cut off the last char in this case, right?
And the substring way of cutting the last char is very wasteful. It should to be done before the SB.toString() to avoid the unnecessary temp strings.